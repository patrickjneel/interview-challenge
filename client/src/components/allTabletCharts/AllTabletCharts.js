import React from 'react';
import TabletPieCost from '../tabletCharts/TabletPieChartClick';
import TabletPieSearch from '../tabletCharts/TabletPieChartSearch';
import PropTypes from 'prop-types';

const AllTabletCharts = ({ data, cost, search }) => {
  return (
    <div>
      <h1 className='event-desc tablet-view'>Tablet Charts</h1>
      <h3 className='event-desc'>Day: {data[0].dayOfWeek}</h3>
      <h3 className='event-desc'>Event: {data[0].campaignName}</h3>
      <h3 className='event-desc'>Device: {data[0].device}</h3>
      <TabletPieCost cost={cost} />
      <TabletPieSearch search={search} />
    </div>
  );
};

export default AllTabletCharts;

AllTabletCharts.propTypes = {
  data: PropTypes.array,
  cost: PropTypes.array,
  hours: PropTypes.array,
  search: PropTypes.array
};
