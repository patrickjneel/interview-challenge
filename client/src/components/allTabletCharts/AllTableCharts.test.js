import React from 'react';
import { shallow } from 'enzyme';
import AllTabletCharts from './AllTabletCharts';

describe('All Tablet Charts Tests', () => {

  it('should match snapshot', () => {
    const mockData = [0];
    const mockCost = [];
    const mockSearch = [];
    const mockHours = [];
    const renderedAllTabletCharts = shallow(
      <AllTabletCharts 
        data={mockData}
        cost={mockCost}
        search={mockSearch}
        hours={mockHours}
      />
    );

    expect(renderedAllTabletCharts).toMatchSnapshot();
  });
});
