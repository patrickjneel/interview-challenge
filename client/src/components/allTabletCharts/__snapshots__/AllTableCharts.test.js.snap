// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Tablet Charts Tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllTabletCharts
    cost={Array []}
    data={
        Array [
            0,
          ]
    }
    hours={Array []}
    search={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="event-desc tablet-view"
>
          Tablet Charts
</h1>,
        <h3
          className="event-desc"
>
          Day: 
</h3>,
        <h3
          className="event-desc"
>
          Event: 
</h3>,
        <h3
          className="event-desc"
>
          Device: 
</h3>,
        <TabletBarCost
          cost={Array []}
/>,
        <TabletPieSearch
          hours={Array []}
          search={Array []}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Tablet Charts",
          "className": "event-desc tablet-view",
        },
        "ref": null,
        "rendered": "Tablet Charts",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Day: ",
            undefined,
          ],
          "className": "event-desc",
        },
        "ref": null,
        "rendered": Array [
          "Day: ",
          undefined,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Event: ",
            undefined,
          ],
          "className": "event-desc",
        },
        "ref": null,
        "rendered": Array [
          "Event: ",
          undefined,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Device: ",
            undefined,
          ],
          "className": "event-desc",
        },
        "ref": null,
        "rendered": Array [
          "Device: ",
          undefined,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cost": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "hours": Array [],
          "search": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="event-desc tablet-view"
>
            Tablet Charts
</h1>,
          <h3
            className="event-desc"
>
            Day: 
</h3>,
          <h3
            className="event-desc"
>
            Event: 
</h3>,
          <h3
            className="event-desc"
>
            Device: 
</h3>,
          <TabletBarCost
            cost={Array []}
/>,
          <TabletPieSearch
            hours={Array []}
            search={Array []}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tablet Charts",
            "className": "event-desc tablet-view",
          },
          "ref": null,
          "rendered": "Tablet Charts",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Day: ",
              undefined,
            ],
            "className": "event-desc",
          },
          "ref": null,
          "rendered": Array [
            "Day: ",
            undefined,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Event: ",
              undefined,
            ],
            "className": "event-desc",
          },
          "ref": null,
          "rendered": Array [
            "Event: ",
            undefined,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Device: ",
              undefined,
            ],
            "className": "event-desc",
          },
          "ref": null,
          "rendered": Array [
            "Device: ",
            undefined,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cost": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "hours": Array [],
            "search": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
